### PR TITLE
Generate a random ssh password per kluster

### PR DESCRIPTION
**Pull request by databus23**
Instead of hardcoding a global password into the ignition template we now generate a random 12 char password for each kluster and use that in the ignition templates for the kluster nodes.
Existing klusters that don’t have this are staying on the old hard coded value for now.

This password is not exposed anyware at the moment. Its meant as a diagnostic backdoor during the techpreview. We can think about exposing this password to the enduser in the future.

Manually tested that this actually works.

### CODE REVIEW

File: charts/kube-master/templates/secrets.yaml [hunk 1]
```diff
@@ -18,9 +18,9 @@ data:
   openstack-router-id: {{ required "missing openstack-router-id" .Values.openstack.routerID | b64enc }}
   openstack-project-id: {{ required "missing openstack-project-id" .Values.openstack.projectID | b64enc }}
   openstack-region: {{ required "missing openstack-region" .Values.openstack.region | b64enc }}
-
   bootstrapToken: {{ required "missing boostrapToken" .Values.bootstrapToken | b64enc }}
   token.csv: {{ include (print $.Template.BasePath "/_token.csv.tpl") . | b64enc }}
+  node-password: {{ required "missing node-password" .Values.nodePassword | b64enc }}
```
Comment (BugRoger) [inline charts/kube-master/templates/secrets.yaml]: Could you also update the chart version to `master-1.9.0-kubernikus.1`? Even though this is informational only, we can see why Kluster is using which chart version.

```
🐢  > helm ls                                                                                                                                                                                                                                                                                                         0 ↵
NAME                                  	REVISION	UPDATED                 	STATUS  	CHART                         	NAMESPACE
test2-8b25871959204ff1a27605b7bcf873f7	1       	Tue Jan  9 11:34:00 2018	DEPLOYED	kube-master-1.9.0-kubernikus.0	kubernikus
v6-5d725ddf97664a16b011e8a8dd75772b   	1       	Fri Jan 12 11:36:08 2018	DEPLOYED	kube-master-1.9.0-kubernikus.0	kubernikus-michi
```

File: pkg/templates/ignition.go [hunk 2]
```diff
@@ -61,6 +63,26 @@ func (i *ignition) GenerateNode(kluster *kubernikusv1.Kluster, secret *v1.Secret
 		return nil, err
 	}
 
+	//this is the old default for backwards comptibility with clusters that don't have a passwort generated
+	//TODO: Remove once all klusters are upgraded
+	passwordHash := "$6$rounds=1000000$aldshc,xbneroyw$I756LN/FtceE1deC2H.tGeSdeeelaaZWRwzmbEuO1SANf7ssyPjnbQjlW/FcMvWGUGrhF64tX9fK0abE/4oQ80"
```
Comment (BugRoger) [inline pkg/templates/ignition.go]: wth? 🤣

File: pkg/templates/ignition.go [hunk 3]
```diff
@@ -61,6 +63,26 @@ func (i *ignition) GenerateNode(kluster *kubernikusv1.Kluster, secret *v1.Secret
...
+		//generate 16 byte random salt
+		salt, err := goutils.Random(16, 32, 127, true, true)
+		if err != nil {
+			return nil, fmt.Errorf("Unable to generate random salt: %s", err)
+		}
+		//We crank up the heat to 1 million rounds of hashing for this password (default 5000)
+		//Reason for this is we expose the resulting hash in the metadata service which is not very secure.
+		//It takes about 500ms on my workstation to compute this hash. So this means login to a node is also
+		// delayed for about a second which should be ok as this password is only meant as a last resort.
+		passwordHash, err = passwordCrypter.Generate(nodePassword, append([]byte("$6$rounds=1000000$"), salt...))
```
Comment (BugRoger) [inline pkg/templates/ignition.go]: How long and complicated is the resulting password? It often needs to be entered in the crappy VPC console without copy-paste...

File: pkg/templates/node_1.7.go [hunk 4]
```diff
@@ -6,7 +6,7 @@ var Node_1_7 = `
 passwd:
   users:
     - name:          core
-      password_hash: xyTGJkB462ewk
+      password_hash: {{ .LoginPassword }}
       ssh_authorized_keys: 
         - "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAvFapuevZeHFpFn438XMjvEQYd0wt7+tzUdAkMiSd007Tx1h79Xm9ZziDDUe4W6meinVOq93MAS/ER27hoVWGo2H/vn/Cz5M8xr2j5rQODnrF3RmfrJTbZAWaDN0JTq2lFjmCHhZJNhr+VQP1uw4z2ofMBP6MLybnLmm9ukzxFYZqCCyfEEUTCMA9SWywtTpGQp8VLM4INCxzBSCuyt3SO6PBvJSo4HoKg/sLvmRwpCVZth48PI0EUbJ72wp88Cw3bv8CLce2TOkLMwkE6NRN55w2aOyqP1G3vixHa6YcVaLlkQhJoJsBwE3rX5603y2KjOhMomqHfXxXn/3GKTWlsQ== michael.j.schmidt@gmail.com"
```
Comment (BugRoger) [inline pkg/templates/node_1.7.go]: Remove this key

File: charts/kube-master/templates/secrets.yaml (hunk 1 above)
Comment (databus23) [reply charts/kube-master/templates/secrets.yaml]: I can do that, yes

File: pkg/templates/ignition.go (hunk 2 above)
Comment (databus23) [reply pkg/templates/ignition.go]: Thats just the old password  salted and hashed 1 million times with sha512. Nothing special.
image

File: pkg/templates/ignition.go (hunk 3 above)
Comment (databus23) [reply pkg/templates/ignition.go]: Its 12 chars and only characters and numbers. Its ok, I tested it.

File: pkg/templates/node_1.7.go (hunk 4 above)
Comment (databus23) [reply pkg/templates/node_1.7.go]: I will with the ssh public key PR I'm working on.

**Pull request merged**